whois-api/src/main/java/net/ripe/db/whois/api/apiKey/BearerTokenExtractor.java

```diff
@@ -46,25 +49,25 @@ public OAuthSession extractBearerToken(final HttpServletRequest request, final S
 
     private OAuthSession getOAuthSession(final String bearerToken, final String apiKeyId) {
         if(StringUtils.isEmpty(bearerToken)) {
-            return new OAuthSession(apiKeyId);
+            return new OAuthSession(apiKeyId, this.environment);
         }
 
         try {
             final SignedJWT signedJWT = SignedJWT.parse(StringUtils.substringAfter(bearerToken, "Bearer "));
 
             if(!verifyJWTSignature(signedJWT)) {
               LOGGER.debug("JWT signature verification failed for {}", apiKeyId);
-              return new OAuthSession(apiKeyId);
+              return new OAuthSession(apiKeyId, this.environment);
             }
             
             //TODO[MA]: remove when apiKeyId is available from api registry call
-            return OAuthSession.from(new ObjectMapper().readValue(signedJWT.getPayload().toString(), OAuthSession.class), apiKeyId);
+            return OAuthSession.from(new ObjectMapper().readValue(signedJWT.getPayload().toString(), OAuthSession.class), apiKeyId, this.environment);
```

Comment:
@maggarwal13 I am unsure if this is OK
My way of thinking is that we are now identifying a session by its environment as well. 
keycloak is using same env for our PROD, Training, TEST, RC so for them the session is the same for all of those environment...but not for us.
So for me make sense to add environment into the OAuthSession for identifying our env from the session that Keycloak is providing. But maybe it is overthinking and no clear?